### DISCUSSION TITLE
Submit data via an API call ?

### DISCUSSION THREAD

**Discussion by just-alexis** [sulu/SuluFormBundle#394] [Q&A]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | no
| 2.5 |
| 2.6.4 |

#### Actual Behavior

I don't know if I'm missing a configuration or something else.
But it seems that there are only 2 possibilities, either send an email or save the responses to the database.

#### Expected Behavior

I would like to make an API call and send the form elements with POST or GET.
Can I do a specific development to do this?

**Comment by alexander-schranz** [comment]
@just-alexis you can use `Ajax` call to submit data via POST to the same URL as the Page.

Via `GET` an not submit data as the Form Bundle only supports POST based data. And based on Http Specification you shouldn't use GET for that.

If you want to know how the Submit handling works check the https://github.com/sulu/SuluFormBundle/blob/12d2df14f2af7ea2ec3e0c3cf1f1638c71325b20/Event/RequestListener.php#L70

> But it seems that there are only 2 possibilities, either send an email or save the responses to the database.

Not sure what you mean here. If you don't disable saving data it always saved to the database. If you configure notify email address, notify are alway send. So can have both saving to database and sending emails. Emails can be send to the notify (admin) email and to the website visitor (if email field exists).

**Comment by alexander-schranz** [comment]
An example of JS based submittion via Dropzone can also be found here in the docs: https://github.com/sulu/SuluFormBundle/blob/2.5/Resources/doc/dropzone.md#2-form-submit-handling-with-multiple-dropzones

**Comment by just-alexis** [comment]
> you can use Ajax call to submit data via POST to the same URL as the Page.

I'm not sure I understand correctly. With javascript, I need to intercept the form submit event, make an AJAX call to my external API, then let the default behavior continue ? That's right ?

**Comment by alexander-schranz** [comment]
Yes that is one way of doing it. The frontend is all in your control so you can to it the way you want it. Aslong as you send it in the format the Form Bundle expect it it doesnt care if its a normal submit or a ajax, curl or other call.

**Comment by just-alexis** [comment]
Thank you for your quick help, it is very pleasant. I will try to implement this solution.

I have another question that is unrelated to the original topic but maybe you can help me.
I am in a french company so the localization of the admin interface is "FR". For the form bundle, it seem like the translation is not available, so i see sulu variable name everywhere in the "sulu_form.forms" tab menu.
How can i add my own translation for this bundle ?
![image](https://github.com/user-attachments/assets/ddb52d36-2141-452c-a41a-7e7e0edcf1e0)

**Comment by alexander-schranz** [comment]
FR is a none core language. You can download community translations like documented in the Sulu Getting Started Documentation via `bin/console sulu:admin:download-language fr`

**Comment by alexander-schranz** [comment]
Translations are done by the community here: https://sulu.crowdin.com/

**Comment by just-alexis** [comment]
Thx !
